gradio/routes.py

```diff
@@ -352,11 +353,13 @@ def api_info():
         @app.get("/config/", dependencies=[Depends(login_check)])
         @app.get("/config", dependencies=[Depends(login_check)])
         def get_config(request: fastapi.Request):
-            config = copy.deepcopy(app.get_blocks().config)
-            root_path = route_utils.get_root_url(request)[: -len("/config")]
-            config["root"] = root_path
-            config = add_root_url(config, root_path)
-            return config
+            config = app.get_blocks().config
+            root = route_utils.get_root_url(request)[: -len("/config")]
+            previous_root = config.get("root", None)
+            if previous_root is None or previous_root != root:
+                config["root"] = root
+                config = add_root_url(config, root, previous_root)
+            return ORJSONResponse(content=config)
```

Comment:
Another change is that explicitly serialize the config using `ORJSONResponse` here. While `ORJSONResponse` is already the default serializer, if you return a dictionary-like from a fastapi route, it first tries to copy/cast it to a dictionary, which causes issues if the dictionary has non-pickle-able values.

So this was previously a bug (if a user tried to access the /config route and the config contained non-pickle-values, it would error out)